src/Features/LanguageServer/Protocol/ExternalAccess/Razor/SemanticTokensRangesHandler.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.ExternalAccess.Razor;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(SemanticTokensRangesHandler)), Shared]
+[Method(SemanticRangesMethodName)]
+internal class SemanticTokensRangesHandler : ILspServiceRequestHandler<SemanticTokensRangesParams, SemanticTokens>
+{
+    public const string SemanticRangesMethodName = "textDocument/semanticTokens/ranges";
```

Comment:
is this a well known lsp method, and this stirng matches that?  or is this an extension we're experiemnting with.  if the latter, we should rename to indicate as much.

Reply:
It was approved in the LSP meeting and we would be eventually adding it. Currently at the experimental stage

Reply:
@CyrusNajmabadi / @dibarbet -- Do you know if there is a convention for names that are currently used to interact between Razor/Roslyn?

Reply:
The two previous names used between Razor/Roslyn have been prefixed with `roslyn/`:

https://github.com/dotnet/razor/blob/f8e2cc5062e278227a9f915167189b156e1bea0e/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPConstants.cs#L24-L26

Reply:
I added a custom endpoint with `roslyn/` prefix, similar to the other two shown in [RazorLSPConstants.cs](https://github.com/dotnet/razor/blob/f8e2cc5062e278227a9f915167189b156e1bea0e/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPConstants.cs#L24-L26)